lib/server/controllers/logs.controller.js

```diff
@@ -223,6 +223,42 @@ const listTagsByLogId = async (request, response, next) => {
     }
 };
 
+/**
+ * Lists all environmemts associated with a log.
+ *
+ * @param {Object} request The *request* object represents the HTTP request and has properties for the request query
+ *                         string, parameters, body, HTTP headers, and so on.
+ * @param {Object} response The *response* object represents the HTTP response that an Express app sends when it gets an
+ *                          HTTP request.
+ * @param {Function} next The *next* object represents the next middleware function which is used to pass control to the
+ *                        next middleware function.
+ * @returns {undefined}
+ */
+const listEnvironmentsByLogId = async (request, response, next) => {
+    const value = await dtoValidator(GetLogDto, request, response);
+    if (!value) {
+        return;
+    }
+
+    const log = await new GetLogUseCase()
```

Comment:
UseCases should not be used in new code. 
You should create a method in EnvironmentService to fetch env by logId.